x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/CloseJobRequest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.ml;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CloseJobRequest extends ActionRequest {
+
+    public static final String ALL_JOBS = "_all";
+
+    private List<String> jobIds = new ArrayList<>();
+    private TimeValue timeout;
+    private boolean force;
+    private boolean allowNoJobs = true;
+
+    public static CloseJobRequest closeAllJobsRequest(){
+        return new CloseJobRequest(ALL_JOBS);
+    }
+
+    public CloseJobRequest(String jobId) {
```

Comment:
Shall we add a constructor that takes a `List<String>` and make `jobIds` final? As the job IDs is the required part of the request, I think replacing the setter with a constructor guides the user better towards making a valid request.

Reply:
Actually, even better, we can have a single constructor that takes `(String... jobIds)`.

Reply:
We can do that, I just tested and doing a close with an empty string is a valid request, it just does not do anything. This could indeed cause confusion and possibly latent bugs in a consumer's code base.